### DISCUSSION TITLE
Settings not saved

### DISCUSSION THREAD

**Discussion by andriuskr** [emsesp/EMS-ESP32#1011] [General]
Flashed new ESP32 with newest firmware, connected to boiler. Everything seems fine and working, until reboot. After reboot ems-esp reverst to factory defaults. 
If i try to save configuration, i get empty file

**Comment by proddy** [comment]
which version?

**Comment by andriuskr** [comment]
3.4.2

**Comment by MichaelDvP** [comment]
Same issue as #614? Try dev-build (3.5.0b)  or flash 3.4.1 and update via upload. What flashtool have you used?

**Comment by andriuskr** [comment]
Unable to upload 3.5.0b firmware. Getting internal server error with different browsers

**Comment by oliverm01** [comment]
Hello! I had exactly the same issue here after flashing 3.4.2 and had my network settings as a suspect first, because the esp32 didn't connect to the network after set and reboot. But the settings file had not been saved.

An firmware upload to 3.5.0b from a local storage (download first) worked and fixed this issue here finally.

**Comment by MichaelDvP** [comment]
> Unable to upload 3.5.0b firmware. Getting internal server error with different browsers

See #629

**Comment by bbqkees** [comment]
Indeed with 3.4.2 on a new Gateway settings are not persistent. Got quite a number of emails about that already.
I test each Gateway before shipment, but I did not test making changes to settings as i ship them out in default state.

For those who ordered a Gateway in the past few weeks and it has 3.4.2 on it, you can load 3.5.0 via the web interface and then it will work. Other option is to load 3.4.1 via the web and then upload 3.4.2.
If loading 3.5.0 fails with error 500/server error please use the USB flashing method with the new EMS-ESP flash tool to flash the Gateway. You have to open the device to access the USB connector. See:
https://bbqkees-electronics.nl/wiki/gateway/firmware-update-and-downgrade.html#uploading-the-firmware-via-ems-esp-flasher-flashtool

**Comment by Wuehli** [comment]
> Got quite a number of emails about that already

+1 😄

**Comment by andriuskr** [comment]
With usb flashing of 3.5.0b6 sucess, but in the end logs shows erros. And ESP does not boot.

<img width="729" alt="image" src="https://user-images.githubusercontent.com/13429780/195013992-6f61082f-b0b0-41d2-a2cd-a8c3fc652dca.png">

**Comment by MichaelDvP** [comment]
What flasher do you have used? The flashers for tasmota and esphome  use a to small partition for ems-esp v.3.5, 
Use esptool from commandline or try [this](https://github.com/MichaelDvP/EMS-ESP_Flasher/releases)

**Comment by andriuskr** [comment]
> What flasher do you have used? The flashers for tasmota and esphome use a to small partition for ems-esp v.3.5,
> Use esptool from commandline or try [this](https://github.com/MichaelDvP/EMS-ESP_Flasher/releases)

No difference. Can flash 3.4.3 version (settings not saved), but 3.5.0b6 gives errors after flashing.
<img width="741" alt="image" src="https://user-images.githubusercontent.com/13429780/195064401-0924be76-b196-43d8-8bdd-aaa761bc5262.png">

**Comment by bbqkees** [comment]
Which module are you trying to flash?

**Comment by MichaelDvP** [comment]
- in v3.4.3 the settings not saved error is fixed.
- have you tried to powercycle the module after flashing?

**Comment by andriuskr** [comment]
> Which module are you trying to flash?
![IMG_1652](https://user-images.githubusercontent.com/13429780/195113900-3783c832-31a8-4001-a7e3-4256ada823df.jpeg)
![IMG_1651](https://user-images.githubusercontent.com/13429780/195113930-52c57deb-32f4-46b0-b47b-902f39a3e1af.jpeg)

ESP 32 D1 MINI.

**Comment by proddy** [comment]
is it working now?

**Comment by andriuskr** [comment]
> is it working now?

No, settings still not saved

**Comment by MichaelDvP** [comment]
I've checked the 3.4.3 and can confirm it is NOT working. The fix is placed at wrong place after trying to read from fs and abort, it should be first in writeToFS before _fs->open().

@proddy Please make a fixed 3.4.4. 

@andriuskr up/downgrade (usb or OTA or web upload) to <= 3.4.1 or 3.5.0bx and configure network, then you can change to any version.

**Comment by MichaelDvP** [comment]
@andriuskr you can use the new release 3.4.4. https://github.com/emsesp/EMS-ESP32/releases/tag/v3.4.4

**Comment by andriuskr** [comment]
> 
Now it is working as expected. Thanks
